packages/tpetra/core/src/Tpetra_Details_DualViewUtil.hpp

```diff
@@ -42,19 +42,15 @@ makeDualViewFromOwningHostView
   using execution_space = typename DeviceType::execution_space;
   using dual_view_type = Kokkos::DualView<ElementType*, DeviceType>;
 
-  if (dv.extent (0) == hostView.extent (0)) {
-    // We don't need to reallocate the device View.
-    dv.clear_sync_state ();
-    dv.modify_host ();
-    dv.h_view = hostView;
-    dv.sync_device ();
-  }
-  else {
-    auto devView = Kokkos::create_mirror_view (DeviceType (), hostView);
-    // DEEP_COPY REVIEW - DEVICE-TO-HOSTMIRROR
-    Kokkos::deep_copy (execution_space(), devView, hostView);
-    dv = dual_view_type (devView, hostView);
-  }
+  typename Kokkos::DualView<ElementType*, DeviceType>::t_dev devView;
+  if (dv.extent (0) == hostView.extent (0))
+    devView = dv.view_device();
+  else
+    devView = Kokkos::create_mirror_view (DeviceType (), hostView);
+  // DEEP_COPY REVIEW - DEVICE-TO-HOSTMIRROR
+  Kokkos::deep_copy (execution_space(), devView, hostView);
+  dv = dual_view_type (devView, hostView);
+  execution_space().fence();
```

Comment:
This fence is new but seems necessary. The constructor will not fence and the `deep_copy` is asynchronous. The function doesn't take an execution space instance argument so it should be synchronous.